anndata/core/alignedmapping.py

```diff
@@ -81,6 +78,15 @@ def _validate_value(self, val: V, key: str) -> None:
                 raise IndexError()  # Maybe not index error
         except AttributeError:
             pass
+        if not self._allow_df and isinstance(val, pd.DataFrame):
+            was_heterogeneous = val.dtypes.nunique() != 1
+            val = val.to_numpy()
```

Comment:
Hmm. I think I'd say go for a reference and if it's confusing someone will open an issue. I think that'd feel more consistent to me.

More generally, when we do `adata.raw = adata` we make a copy. Is that counter intuitive if we otherwise assign references?